app/components/listing/ResourceInfos.jsx

```diff
@@ -30,37 +30,53 @@ ResourceCategories.propTypes = {
 };
 
 
+// Given an address object, returns a React DOM element that displays the
+// address on up to three lines:
+//
+//   <name>
+//   <address_1>, <address_2>
+//   <city>, <state_province>, <postal_code>
+//
+// Example 1:
+//
+//   123 Fourth Street
+//   San Francisco, CA, 94115
+//
+// Example 2:
+//
+//   SF Headquarters
+//   567 Eighth Street, Suite 201
+//   San Francisco, CA, 94115
+//
 function buildLocation(address) {
-  let line1 = '';
-  let line2 = '';
+  const lines = [];
 
   if (address) {
-    if (address.address_1) {
-      line1 += address.address_1;
-    }
-
-    if (address.address_2) {
-      line1 += `, ${address.address_2}`;
-    }
-
-    if (address.city) {
-      line2 += address.city;
-    }
-
-    if (address.state_province) {
-      line2 += `, ${address.state_province}`;
-    }
-
-    if (address.postal_code) {
-      line2 += `, ${address.postal_code}`;
-    }
+    const fieldsOnEachLine = [
+      ['name'],
+      ['address_1', 'address_2'],
+      ['city', 'state_province', 'postal_code'],
+    ];
+    fieldsOnEachLine.forEach(fields => {
+      const line = fields.map(field => address[field])
+                     .filter(value => value)
+                     .join(', ');
+      if (line.length > 0) {
+        lines.push(line);
+      }
+    });
   }
 
   return (
     <span>
-      {line1}
-      <br />
-      {line2}
+    {
+      lines.map((line, i) =>
+        <Fragment key={i}>
```

Comment:
Generally, you want to avoid using index as a key. The key is usually a unique identifier to to the piece of data that element holds, whereas an index isn't necessarily linked to that piece of data forever. Usually you would use something like an id. In this case, it honestly isn't really going to matter - it's more of a problem with long lists. Just more of a general FYI since you mentioned you were newer to this stuff. :)

Reply:
In react, you can return an array of elements and it knows how to render them. So, you could avoid mapping twice by doing something like:

```
function buildLocation(address) {
  if (!address) return;

  const fieldsOnEachLine = [
    ['name'],
    ['address_1', 'address_2'],
    ['city', 'state_province', 'postal_code'],
  ];

  return fieldsOnEachLine.map(fields => {
    const line = fields.map(field => address[field])
                   .filter(Boolean)
                   .join(', ');

    if (line.length > 0) {
      const key = fields.join('-');

      return (
        <Fragment key={key}>
          <span>{line}</span>
          <br />
        </Fragment>
      );
    }
  });
}
```

Again, given that this is a list of at most 3, it doesn't really matter and feel free to just leave it the way you have it now.

Reply:
Awesome! Thanks for the info about keys, and I love `filter(Boolean)`. Will make these changes